rpc: avoid modifying shared TLS config to prevent data race

cthumuluru-crdb opened the pull request:
The certificate manager initializes a new TLS configuration instance and caches it for reuse. Modifying the cached TLS configuration introduced a data race. Fixed the data race and unskipped the test that was skipped in #139189.

Fixes: #139134
Epic: none
Release note: none

cthumuluru-crdb committed:
rpc: avoid modifying shared TLS config to prevent data race

The certificate manager initializes a new TLS configuration instance and caches it
for reuse. Modifying the cached TLS configuration introduced a data race.

Fixes: #139134
Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139198)
<!-- Reviewable:end -->

tbg reviewed (approved):
Thank you and sorry for slipping this in 🥲

tbg commented:
Don't forget to also unskip the test (see https://github.com/cockroachdb/cockroach/pull/139189)

You'll need to `git fetch https://github.com/cockroachdb/cockroach master && git rebase FETCH_HEAD` (to stack your commit on top of master, so you can see #139189), then make a commit that unskips the test, then push.

tbg reviewed (changes requested):
needs test unskip

cthumuluru-crdb committed:
server: unskip TestDRPCSelectQuery

TestDRPCSelectQuery was skipped under race in #139189 due to failures caused by data races.
This PR re-enables the test.

Fixes: #139134
Release note: None
Epic: None

cthumuluru-crdb reviewed (commented):
No worriers. Unskipped the test now. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139198#-:-OGkNsLAEYiSG8SFqvfe:b-q7unot)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg reviewed (approved):
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139198)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @cthumuluru-crdb)



<!-- Sent from Reviewable.io -->

cthumuluru-crdb commented:
bors r+

tbg commented:
blathers backport 25.1

cthumuluru-crdb commented:
blathers backport 25.1
